lib/Mojolicious/Plugin/AssetPack/Pipe/Export.pm

```diff
@@ -0,0 +1,92 @@
+
+package Mojolicious::Plugin::AssetPack::Pipe::Export;
+
+use Mojo::Base 'Mojolicious::Plugin::AssetPack::Pipe';
+use Mojo::Util 'spurt';
+
+use File::Basename 'dirname';
+use File::Path 'make_path';
+use File::Spec;
+
+has 'export_dir';
+has 'use_checksum_subdir' => 1;
+
+sub process {
+	my ($self, $assets) = @_;
+
+	my $dir = $self->export_dir // $ENV{'MOJO_ASSETPACK_EXPORT_DIR'};
+	unless ( defined $dir and -e -w -d $dir ) {
+		die "Missing or inaccesable export base directory '$dir'" 
+	}
+
+	$assets->each(sub {
+		my $asset = shift;
+
+		my $path;
+		if ( $self->use_checksum_subdir ) {
```

Comment:
What my question really is: Why support both cases?

Also, you could use ```$self->assetpack->route``` and create the directory structure from the path defined by the route object.